tests/integration/test_agentd/test_agentd_multi_server/test_agentd_multi_server.py

```diff
@@ -0,0 +1,204 @@
+        - Agent with keys. Verify that the first server is available, but it disconnects, and the second and
+          third servers are not responding. The agent on disconnection should try the second and third servers
+          and go back finally to the first server.
+
+    input_description: An external YAML file (wazuh_conf.yaml) includes configuration settings for the agent.
+                       Different test cases are found in the test module and include parameters for
+                       the environment setup, the requests to be made, and the expected result.
+
+    expected_output:
+        - r'Requesting a key from server'
+        - r'Valid key received'
+        - r'Trying to connect to server'
+        - r'Connected to enrollment service'
+        - r'Received message'
+        - r'Unable to connect to enrollment service at'
+
+    tags:
+        - simulator
+        - ssl
+        - keys
+    '''
+
+    # Servers paremeters
+    remoted_server_addresses = ["127.0.0.0","127.0.0.1","127.0.0.2"]
+    remoted_server_ports = [1514,1516,1517]
+    remoted_servers = [None,None,None]
+    
+    # Stop target Agent
+    control_service('stop')
+
+    # Configure keys
+    if(test_metadata['SIMULATOR_MODES']['AUTHD'] == 'ACCEPT'):
+        authd_server = AuthdSimulator(server_ip = remoted_server_addresses[0], mode = test_metadata['SIMULATOR_MODES']['AUTHD'])
+        authd_server.start()
+    else:
+        if(test_metadata['SIMULATOR_MODES']['AUTHD_PREV_MODE'] == 'ACCEPT'):
+            authd_server = None
+            add_custom_key()
+
+    # Start target Agent
+    control_service('start')
+    
+    # Start Remoted Simulators
+    for i in range(len(remoted_server_addresses)):
+        if(test_metadata['SIMULATOR_MODES'][i] != 'CLOSE'):
+            remoted_servers[i] = RemotedSimulator(protocol = test_metadata['PROTOCOL'], server_ip = remoted_server_addresses[i], 
+                                        port = remoted_server_ports[i], mode = test_metadata['SIMULATOR_MODES'][i])
+            remoted_servers[i].start()
+
+    # Start FileMonitor
+    log_monitor = FileMonitor(WAZUH_LOG_PATH)
+
+    # Iterate the servers from which we are expecting logs
+    for server in range(len(test_metadata['LOG_MONITOR_STR'])):
+        # Iterate the patterns expected for server
+        for pattern in range(len(test_metadata['LOG_MONITOR_STR'][server])):
+            # Build regex from expected pattern
+            regex, values = get_regex(test_metadata['LOG_MONITOR_STR'][server][pattern],
+                              remoted_server_addresses[server],
+                              remoted_server_ports[server])
+            # Look for expected log
+            log_monitor.start(callback=callbacks.generate_callback(regex,values))
+            assert (log_monitor.callback_result != None), regex
+
+    # Shutdown simulators
+    for i in range(len(remoted_servers)):
```

Comment:
Same, this could be in the same fixture after a yield.

Reply:
Fixed in https://github.com/wazuh/wazuh/pull/18353/commits/ecf23d03eafffc97b67a2d364324e916eb50d143